src/NHibernate/Persister/Collection/OneToManyPersister.cs

```diff
@@ -66,8 +66,18 @@ protected override SqlCommandInfo GenerateDeleteString()
 		{
 			var update = new SqlUpdateBuilder(Factory.Dialect, Factory)
 				.SetTableName(qualifiedTableName)
-				.AddColumns(KeyColumnNames, "null")
-				.SetIdentityColumn(KeyColumnNames, KeyType);
+				.AddColumns(JoinColumnNames, "null");
+
+			if (CollectionType.UseLHSPrimaryKey)
+			{
+				update.SetIdentityColumn(KeyColumnNames, KeyType);
+			}
+			else
+			{
```

Comment:
@JKeyser Why do we need this join?

It fails on SQLite, because it does not support joins at the update statements.